spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-enhanced-ecommerce-tracker.spec.js

```diff
@@ -0,0 +1,296 @@
+/* eslint-env jasmine */
+
+describe('Google Analytics ecommerce tracking', function () {
+  var GOVUK = window.GOVUK
+  var searchResultsParentEl
+  var searchResults
+  var searchResultsHeading
+  var resultToBeClicked
+  var onPageLoadExpected
+  var onSearchResultClickExpected
+  var pageViewExpected
+
+  function agreeToCookies () {
+    GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+  }
+
+  beforeEach(function () {
+    searchResultsHeading = document.createElement('h2')
+    searchResultsHeading.id = 'js-result-count'
+    searchResultsHeading.innerHTML = '12,345 results'
+
+    searchResultsParentEl = document.createElement('div')
+    searchResultsParentEl.setAttribute('data-ga4-ecommerce', '')
+    searchResultsParentEl.setAttribute('data-search-query', 'test-search-query')
+    searchResultsParentEl.setAttribute('data-module', 'ga4-enhanced-ecommerce-tracker')
+    searchResultsParentEl.setAttribute('data-ecommerce-variant', 'test-ecommerce-variant')
+    searchResultsParentEl.setAttribute('data-list-title', 'test-list-title')
+    searchResultsParentEl.setAttribute('data-ecommerce-start-index', '1')
+
+    searchResults = document.createElement('div')
+    searchResults.innerHTML =
+      '<a data-ecommerce-path="/coronavirus" data-ecommerce-row="1" data-ecommerce-index="1" href="/coronavirus">Coronavirus (COVID-19): guidance and support</a>' +
+
+      '<a data-ecommerce-path="/guidance/travel-abroad-from-england-during-coronavirus-covid-19" data-ecommerce-row="1" data-ecommerce-index="2" href="/guidance/travel-abroad-from-england-during-coronavirus-covid-19">Travel abroad from England during coronavirus (COVID-19)</a>' +
```

Comment:
Non blocking comment - it's good to base test data on real data but we can make our test data more readable by shortening things to the minimum i.e. the text and href of these links could be much shorter - as long as they're unique.

Reply:
Cool, I've just pushed up a change that will shorten those test links.